### PR TITLE
Add irecv wrapper in KokkosComm namespace

### PR DESCRIPTION
**Pull request by nicoleavans**
Currently, irecv is only available in the Impl namespace.

### CODE REVIEW

**Review by dssgabriel — DISMISSED** [review]

**Comment by nicoleavans** [comment]
Upon further thought, a function like this could be implemented to more evenly match up with the existing wrappers:

```
template <KokkosView RecvView>
void irecv(const ExecSpace &space, RecvView &sv, int src, int tag, MPI_Comm comm) {
  return Impl::irecv(space, sv, src, tag, comm);
}
```

**Comment by aprokop** [comment]
I'd like this too for #53.

**Review by aprokop — APPROVED** [review]

**Comment by aprokop** [comment]
Also needs `alltoall`

File: src/KokkosComm.hpp [hunk 1]
```diff
@@ -33,6 +34,11 @@ Req isend(const ExecSpace &space, const SendView &sv, int dest, int tag, MPI_Com
   return Impl::isend<SendMode>(space, sv, dest, tag, comm);
 }
 
+template <KokkosView RecvView>
+void irecv(RecvView &sv, int src, int tag, MPI_Comm comm, MPI_Request req) {
```
Comment (dssgabriel) [inline src/KokkosComm.hpp]: Why use `sv` for the receiving view? I think it was previously clearer with `rv` instead.

**Comment by dssgabriel** [comment]
As the `MPI_Alltoall` wrapper has been implemented, consider adding it to this PR as well

File: src/KokkosComm.hpp (hunk 1 above)
Comment (nicoleavans) [reply src/KokkosComm.hpp]: I was trying to more closely match the `recv` wrapper in the file:
```
template <KokkosExecutionSpace ExecSpace, KokkosView RecvView>
void recv(const ExecSpace &space, RecvView &sv, int src, int tag, MPI_Comm comm) {
  return Impl::recv(space, sv, src, tag, comm);
}
```

I agree it is clearer, I can change it back.

Comment (dssgabriel) [reply src/KokkosComm.hpp]: I think you can change the `recv` wrapper to `RecvView &rv` as well then :+1:

File: src/KokkosComm.hpp [hunk 2]
```diff
@@ -33,14 +35,25 @@ Req isend(const ExecSpace &space, const SendView &sv, int dest, int tag, MPI_Com
...
+
 template <CommMode SendMode = CommMode::Default, KokkosExecutionSpace ExecSpace, KokkosView SendView>
 void send(const ExecSpace &space, const SendView &sv, int dest, int tag, MPI_Comm comm) {
   return Impl::send<SendMode>(space, sv, dest, tag, comm);
 }
 
 template <KokkosExecutionSpace ExecSpace, KokkosView RecvView>
-void recv(const ExecSpace &space, RecvView &sv, int src, int tag, MPI_Comm comm) {
-  return Impl::recv(space, sv, src, tag, comm);
+void recv(const ExecSpace &space, RecvView &rv, int src, int tag, MPI_Comm comm) {
```
Comment (cwpearson) [inline src/KokkosComm.hpp:49]: 👍

**Review by cwpearson — APPROVED** [review]

**Review by aprokop — COMMENTED** [review]
If all we do right now is just passing things through, can't we simply do
```c++
  namespace KokkosComm {
    using Impl::recv;
    using Impl::alltoall;
  }
```
Edit: was too late with the review.

**Pull request merged**